src/Metadata/Resource/Factory/DefaultConfigurationResourceMetadataFactory.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+
+/**
+ * Creates a resource metadata from the default configuration.
+ *
+ * @author Beno!t POLASZEK <bpolaszek@gmail.com>
+ */
+final class DefaultConfigurationResourceMetadataFactory implements ResourceMetadataFactoryInterface
+{
+    private $decorated;
+
+    private $options;
+
+    public function __construct(ResourceMetadataFactoryInterface $decorated, array $options)
+    {
+        $this->decorated = $decorated;
+        $this->options = $options;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass): ResourceMetadata
+    {
+        $resourceMetadata = $this->decorated->create($resourceClass);
+        $attributes = $resourceMetadata->getAttributes();
+
+        $baseConfig = array_filter($this->options, function (string $key) {
```

Comment:
Maybe we should just add a continue in the `foreach` instead of having to use `array_filter`?

Reply:
My bad, refactored code which didn't get simplified.